community/kernel/src/main/java/org/neo4j/kernel/impl/proc/ReflectiveProcedureCompiler.java

```diff
@@ -514,12 +524,18 @@ private QualifiedName extractName( Class<?> procDefinition, Method m, String val
 
     private abstract static class ReflectiveBase
     {
-
         final List<FieldInjections.FieldSetter> fieldSetters;
+        private final boolean singleton;
+        protected Object[] args;
+        protected Object cls;
```

Comment:
This will not work!

It makes non-singleton procedures prone to race conditions, and thus not safe for invocations from multiple threads.

With these being fields they will be shared between different invocations of the same procedure, so you can get multiple concurrent invocations using another `cls` or `args` than their own, leading to very strange behaviour.